tools/generate_chain.py

```diff
@@ -28,107 +30,92 @@ def enable_profiler(profile: bool) -> Iterator[None]:
         yield
 
     pr.create_stats()
-    pr.dump_stats("generate-chain.profile")
-
-
-root_path = Path("./test-chain").resolve()
-mkdir(root_path)
-with TempKeyring() as keychain:
-
-    bt = create_block_tools(constants=test_constants, root_path=root_path, keychain=keychain)
-    initialize_logging(
-        "generate_chain", {"log_level": "DEBUG", "log_stdout": False, "log_syslog": False}, root_path=root_path
-    )
-
-    with closing(sqlite3.connect("stress-test-blockchain.sqlite")) as db:
-
-        print("initializing v2 block store")
-        db.execute(
-            "CREATE TABLE full_blocks("
-            "header_hash blob PRIMARY KEY,"
-            "prev_hash blob,"
-            "height bigint,"
-            "in_main_chain tinyint,"
-            "block blob)"
+    pr.dump_stats(f"generate-chain-{counter}.profile")
+
+
+@click.command()
+@click.option("--length", type=int, default=None, required=False, help="the number of blocks to generate")
+@click.option(
+    "--fill-rate",
+    type=int,
+    default=100,
+    required=False,
+    help="the transaction fill rate of blocks. Specified in percent of max block cost",
+)
+@click.option("--profile", is_flag=True, required=False, default=False, help="dump CPU profile at the end")
+@click.option(
```

Comment:
minor thing, but why not have this one as a flag too? `--no-block-refs` ?

Reply:
it probably should have been